Add missing parent IDs, other fixes

jfwood opened the pull request:
Add setting of the parent ID on trackers, fix DynamoDB logic, fix
integer conversion math, fix some logging lines.

Commit:
Add missing parent IDs, other fixes

Add setting of the parent ID on trackers, fix DynamoDB logic, fix
integer conversion math, fix some logging lines.

troylar commented on progressmonitor/__init__.py:
```diff
@@ -244,7 +247,7 @@ def from_json(self, id, j):
             t.with_name(j['TrackerName'], True)
         if 'EstimatedSeconds' in j.keys():
-            t.with_estimated_seconds(j['EstimatedSeconds'], True)
+            t.with_estimated_seconds(int(j['EstimatedSeconds']), True)
```
Just curious, why are we converting to int?

jfwood replied:
This is coming back out of dynamo as a Decimal type. We could convert to float here instead, or just deal with Decimals. They don't play nice with arithmetic involving floats though

troylar commented on progressmonitor/__init__.py:
```diff
@@ -536,13 +569,51 @@ def to_update_item(self):
+        return ue, self.prepare_for_dynamodb(eav)
+
+    @classmethod
+    def prepare_for_dynamodb(cls, data_store):
```
Can we just cast individual properties in the to_update_item method instead of using this method?

Example: 
Change
```
if self.estimated_seconds:
             ue = ue + ', EstimatedSeconds=:est_sec'
             eav[':est_sec'] = self.estimated_seconds
```
to
```
if self.estimated_seconds:
             ue = ue + ', EstimatedSeconds=:est_sec'
             eav[':est_sec'] = str(self.estimated_seconds)
```
I feel like this method is pretty brute force. Because there's only a few fields and there's only a couple float/datetime fields, it seems to fit in the to_update_item. @jfwood

troylar replied:
@jfwood

jfwood replied:
yep, will do shortly.

jfwood commented:
test comment

jfwood commented on progressmonitor/__init__.py:
```diff
@@ -536,13 +569,51 @@ def to_update_item(self):
+
+        return ue, self.prepare_for_dynamodb(eav)
+
+    @classmethod
```
A test comment for @troylar.....this could be a @staticmethod.

jfwood commented:
@troylar, just added a fix for a dynamoDB issue that @xzased found and suggested a fix for. I've also addressed your requested change as well I believe.
